model/task_lifecycle.go

```diff
@@ -1141,3 +1155,45 @@ func checkResetDisplayTask(t *task.Task) error {
 
 	return nil
 }
+
+// if part of single-host task group, reset all tasks if finished and should be reset
+func checkResetTaskGroup(t *task.Task) error {
```

Comment:
I see why this can't be a method on task, but I wish it could be. Is there a way to refactor some of this functionality into methods on either the project or task itself?

Reply:
Yeah I also am not happy that this is not a method on task. Unfortunately I don't think much of this code lends itself to methods, but I can do some general refactoring.